src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/MetadataTemplateServiceImpl.java

```diff
@@ -234,15 +227,46 @@ public MetadataRestriction setMetadataRestriction(Project project, MetadataTempl
 		return metadataRestrictionRepository.save(metadataRestrictionForFieldAndProject);
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	@PreAuthorize("hasPermission(#template, 'canReadMetadataTemplate')")
+	public List<MetadataTemplateField> getPermittedFieldsForTemplate(MetadataTemplate template) {
+		List<MetadataTemplateField> fieldsForTemplate = fieldRepository.getMetadataFieldsForTemplate(template);
+
+		List<MetadataTemplateField> permittedFieldsForCurrentUser = getPermittedFieldsForCurrentUser(
+				template.getProject(), true);
+
+		return fieldsForTemplate.stream()
+				.filter(f -> permittedFieldsForCurrentUser.contains(f))
+				.collect(Collectors.toList());
+	}
```

Comment:
This is the main addition.  Basically it's just reusing the `getPermittedFieldsForCurrentUser` function to filter out which fields from the template can be displayed.